ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/DDLTracerService.java

```diff
@@ -0,0 +1,344 @@
+            } else {
+              String referenceName = schema.columns().get(referencePosition).name();
+              sql.append(String.format(MOVE_AFTER_COLUMN, updateColumn.getName(), referenceName));
+            }
+          }
+          break;
+        case MOVE_AFTER:
+          sql.append(String.format(MOVE_AFTER_COLUMN, updateColumn.getName(), updateColumn.getNewName()));
+          break;
+        case MOVE_FIRST:
+          sql.append(String.format(MOVE_FIRST, updateColumn.getName()));
+          break;
+        default:
+          break;
+      }
+      if (sql.length() > 0) {
+        sql.append(";\\n");
+        schemaSql.append(sql);
+      }
+    }
+    DDLInfo ddlInfo =
+        DDLInfo.of(tableIdentifier, schemaSql.toString(), schemaId, DDLType.UPDATE_SCHEMA.name(), commitTime);
+    insert(ddlInfo);
+  }
+
+  public void commitProperties(TableIdentifier tableIdentifier, Map<String, String> before, Map<String, String> after) {
+    String tableName = TableMetadataUtil.getTableAllIdentifyName(tableIdentifier);
+    Long commitTime = System.currentTimeMillis();
+    StringBuilder setSql = new StringBuilder();
+    StringBuilder unsetSql = new StringBuilder();
+    StringBuilder unsetPro = new StringBuilder();
+    int c = 0;
+    for (String oldPro : before.keySet()) {
+      if (!after.containsKey(oldPro)) {
+        if (c > 0) {
+          unsetPro.append(",").append("\\n");
+        }
+        unsetPro.append(oldPro);
+        c++;
+      }
+    }
+    StringBuilder setPro = new StringBuilder();
+    int c1 = 0;
+    for (String newPro : after.keySet()) {
+      if (!after.get(newPro).equals(before.get(newPro))) {
+        if (c1 > 0) {
+          setPro.append(",").append("\\n");
+        }
+        setPro.append(String.format("'%s'='%s'", newPro, after.get(newPro)));
+        c1++;
+      }
+    }
+    if (setPro.length() > 0) {
+      setSql.append(String.format(ALTER_TABLE, tableName)).append(String.format(SET_PROPERTIES, setPro));
+    }
+    if (unsetPro.length() > 0) {
+      unsetSql.append(String.format(ALTER_TABLE, tableName)).append(String.format(UNSET_PROPERTIES, unsetPro));
+    }
+    if (setSql.length() > 0) {
+      DDLInfo ddlInfo =
+          DDLInfo.of(tableIdentifier, setSql.toString(), null, DDLType.UPDATE_PROPERTIES.name(), commitTime);
+      insert(ddlInfo);
+    }
+    if (unsetSql.length() > 0) {
+      DDLInfo ddlInfo =
+          DDLInfo.of(tableIdentifier, unsetSql.toString(), null, DDLType.UPDATE_PROPERTIES.name(), commitTime);
+      insert(ddlInfo);
+      insert(ddlInfo);
+    }
+  }
+
+  public Integer getCurrentSchemaId(TableIdentifier identifier) {
+    try (SqlSession sqlSession = getSqlSession(true)) {
+      DDLRecordMapper ddlRecordMapper = getMapper(sqlSession, DDLRecordMapper.class);
+      List<Integer> ids = ddlRecordMapper.getCurrentSchemaId(identifier);
+      return ids.get(0) == null ? 0 : ids.get(0);
```

Comment:
> return ids.get(0) == null ? 0 : ids.get(0);
if ids is empty, this will be throw OutOfBoundsException.

try this??
> return ids.size > 0? ids.get(0):0;